demo/api/src/warnings/warning-checker.console.ts

```diff
@@ -0,0 +1,78 @@
+import { BlockInput, FlatBlocks } from "@comet/blocks-api";
+import { DiscoverService } from "@comet/cms-api/lib/dependencies/discover.service";
+import { CreateRequestContext, MikroORM } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityManager, EntityRepository } from "@mikro-orm/postgresql";
+import { Injectable } from "@nestjs/common";
+import { Command, Console } from "nestjs-console";
+import { v5 } from "uuid";
+
+import { Warning } from "./entities/warning.entity";
+
+@Injectable()
+@Console()
+export class WarningCheckerConsole {
+    constructor(
+        private readonly orm: MikroORM,
+        private readonly discoverService: DiscoverService,
+        private readonly entityManager: EntityManager,
+        @InjectRepository(Warning) private readonly warningsRepository: EntityRepository<Warning>,
+    ) {}
+
+    @Command({
+        command: "check-warnings",
+        description: "Checks for warnings",
+    })
+    @CreateRequestContext()
+    async execute(): Promise<void> {
+        // TODO: (in the next PRs) Check if data itself is valid in the database. (Maybe some data was put into database and is not correct or a migration was done wrong)
+
+        for (const rootBlockEntity of this.discoverService.discoverRootBlocks()) {
+            const { metadata, column, block } = rootBlockEntity;
+
+            const queryBuilder = this.entityManager.createQueryBuilder(metadata.className);
+            queryBuilder.select(`id, "${column}"`).from(metadata.tableName);
+            const rootBlocks = (await queryBuilder.getResult()) as Array<{ [key: string]: BlockInput }>;
+
+            for (const rootBlock of rootBlocks) {
+                const blockData = block.blockDataFactory(block.blockInputFactory(rootBlock[column]));
```

Comment:
Here you need to instantiate **all** blocks of the application. Did you some calculations on the expected runtime of this for our larger applications?

Reply:
Good point, I'll do that now.